roles/ui/files/FWO.UI/Shared/FileUpload.razor

```diff
@@ -0,0 +1,194 @@
+            <span class="ms-1">@(userConfig.GetText("import"))</span>
+        </button>
+        @if (ShowAlert)
+        {
+            <div class="alert alert-warning mt-2" role="alert">
+                <span>@userConfig.GetText("E7014")</span>
+            </div>
+        }
+    </div>
+</div>
+@code {
+    private bool ShowAlert { get; set; }
+    InputFileChangeEventArgs? InputFileChangeEventArgs;
+
+    private bool ImportDisabled { get; set; } = true;
+
+    [Parameter]
+    public EventCallback<(Exception, string?)> OnAddAppServerError { get; set; }
+
+    [Parameter]
+    public EventCallback OnImportSuccess { get; set; }
+
+    [Parameter]
+    public ApiConnection ApiConnection { get; set; }
+
+    private List<AppServerType> AppServerTypes = new();
+
+    protected override void OnInitialized()
+    {
+        try
+        {
+            if (userConfig.ModAppServerTypes != "")
+            {
+                AppServerTypes = JsonSerializer.Deserialize<List<AppServerType>>(userConfig.ModAppServerTypes) ?? new();
+            }
+        }
+        catch (Exception) { }
+    }
+
+    private void SingleUpload(InputFileChangeEventArgs e)
+    {
+        ImportDisabled = true;
+        InputFileChangeEventArgs = e;
+        ShowAlert = false;
+
+        if (InputFileChangeEventArgs.FileCount == 0)
+            return;
+
+        if (!e.File.Name.EndsWith(".csv"))
+        {
+            ShowAlert = true;
+            return;
+        }
+
+        ImportDisabled = false;
+    }
+
+    private async Task LoadCSV()
+    {
+        if (ImportDisabled || InputFileChangeEventArgs is null)
+            return;
+
+        using MemoryStream ms = new MemoryStream();
+        await InputFileChangeEventArgs.File.OpenReadStream().CopyToAsync(ms);
+        byte[]? bytes = ms.ToArray();
+
+        string text = System.Text.Encoding.UTF8.GetString(bytes);
+        string[] lines = text.Split('\r');
+
+        foreach (string tmpLine in lines)
+        {
+            string line = tmpLine;
+
+            if (line.StartsWith("\n"))
+                line = line.Remove(0, 1);
+
+            if (IsHeader(line))
+                continue;
+
+            string[] entries = line.Split(';');
+
+            if (entries.Length < 3)
+                continue;
+
+            CSVAppServerImportModel appServer = new()
+                {
+                    AppServerName = entries[0],
+                    AppID = entries[1],
+                    AppServerTyp = entries[2],
+                    AppIPRange = entries[3]
+                };
+
+
+            if (appServer.AppIPRange.TrySplit('-', 1, out string ipEnd))
+            {
+                appServer.AppIPRange = appServer.AppIPRange.Split('-')[0];
+                appServer.AppIPRangeEnd = ipEnd;
+            }
+
+            if (string.IsNullOrEmpty(appServer.AppIPRangeEnd) && !appServer.AppIPRange.TrySplit('/', 1, out string subnet))
+            {
+                appServer.AppIPRange += "/32";
```

Comment:
this needs to be adjusted to not only work for ipv4 (/32) but also for ipv6 (/128)

Reply:
oh right...